internal/metrics/parser.go

```diff
@@ -0,0 +1,60 @@
+// Copyright © 2020 VMware
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/prometheus/common/expfmt"
+)
+
+const prometheusStat = "envoy_http_downstream_cx_active"
+
+func prometheusLabels() []string {
+	return []string{"ingress_http", "ingress_https"}
```

Comment:
Can you use `ENVOY_HTTP_LISTENER` and `ENVOY_HTTPS_LISTENER` here?

Reply:
The ENV vars? No, these strings match labels in the prometheus metrics.

Reply:
I meant [these](https://github.com/projectcontour/contour/blob/master/internal/contour/listener.go#L33). But I suppose these strings are coded in so many places that one more doesn't hurt :)

Resolve this if you want to keep this as is.

Reply:
oh I see. Hmm, possibly. As it's written now we'd get an import cycle by referencing those since the contour package already references the metrics one.

Reply:
maybe can lift the const somewhere else. It feels bad to add more to the shutdown manager file.